server/semantics-service/src/main/java/edu/illinois/ncsa/incore/service/semantics/controllers/TypeController.java

```diff
@@ -47,44 +47,55 @@ public class TypeController {
 
     private final String username;
 
-    private final Authorizer authorizer;
     private final List<String> groups;
 
+    private String endpointUrl = "https://incore.ncsa.illinois.edu/semantics/api/types/";
```

Comment:
Instead of this, could you directly use the hostname if available? and then build the latter part dynamically? I think this would make it more flexible as we have multiple deployment instances for example localhost:8080, incore-dev, and incore. @longshuicy  What are your thoughts on this?

Reply:
Thanks for the suggestion! Looking at data services, they use `System.getenv("SERVICES_URL")`. I just made the change in type controller. But I think this can only be tested when it's deployed to dev/prod.

Reply:
yes we should use environment variables for that